examples/own_dataset/train.py

```diff
@@ -139,48 +166,43 @@ def postprocess_label(label_list):
 
-    train_iter = I.SerialIterator(train, args.batchsize)
-    val_iter = I.SerialIterator(val, args.batchsize,
+    train_iter = iterators.SerialIterator(train, args.batchsize)
+    val_iter = iterators.SerialIterator(val, args.batchsize,
                                 repeat=False, shuffle=False)
 
-    def scaled_abs_error(x0, x1):
-        if isinstance(x0, Variable):
-            x0 = cuda.to_cpu(x0.data)
-        if isinstance(x1, Variable):
-            x1 = cuda.to_cpu(x1.data)
-        if args.scale == 'standardize':
-            scaled_x0 = ss.inverse_transform(cuda.to_cpu(x0))
-            scaled_x1 = ss.inverse_transform(cuda.to_cpu(x1))
-            diff = scaled_x0 - scaled_x1
-        elif args.scale == 'none':
-            diff = cuda.to_cpu(x0) - cuda.to_cpu(x1)
-        return numpy.mean(numpy.absolute(diff), axis=0)[0]
-
-    classifier = L.Classifier(model, lossfun=F.mean_squared_error,
-                              accfun=scaled_abs_error)
+    regressor = Regressor(
+        model, lossfun=F.mean_squared_error,
+        metrics_fun={'abs_error': ScaledAbsError(scale=args.scale, ss=ss)},
+        device=args.gpu)
 
-    if args.gpu >= 0:
-        chainer.cuda.get_device_from_id(args.gpu).use()
-        classifier.to_gpu()
-
-    optimizer = O.Adam()
-    optimizer.setup(classifier)
+    optimizer = optimizers.Adam()
+    optimizer.setup(regressor)
 
     updater = training.StandardUpdater(train_iter, optimizer, device=args.gpu,
                                        converter=concat_mols)
     trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
-    trainer.extend(E.Evaluator(val_iter, classifier, device=args.gpu,
+    trainer.extend(E.Evaluator(val_iter, regressor, device=args.gpu,
                                converter=concat_mols))
     trainer.extend(E.snapshot(), trigger=(args.epoch, 'epoch'))
     trainer.extend(E.LogReport())
-    # Note that scaled errors are reported as (validation/)main/accuracy
-    trainer.extend(E.PrintReport(['epoch', 'main/loss', 'main/accuracy',
+    # Note that original scale absolute errors are reported in
+    # (validation/)main/abs_error
+    trainer.extend(E.PrintReport(['epoch', 'main/loss', 'main/abs_error',
                                   'validation/main/loss',
-                                  'validation/main/accuracy',
+                                  'validation/main/abs_error',
                                   'elapsed_time']))
     trainer.extend(E.ProgressBar())
     trainer.run()
 
+    # --- save regressor's parameters ---
+    protocol = args.protocol
+    save_model_path = os.path.join(args.out, 'model.npz')
+    print('saving trained model to {}'.format(save_model_path))
+    serializers.save_npz(save_model_path, regressor)
+    if args.scale == 'standardize':
+        with open(os.path.join(args.out, 'ss.pkl'), mode='wb') as f:
+            pickle.dump(ss, f, protocol=protocol)
```

Comment:
It seems more natural to me that we implement `serialize` method to `Regressor` which serialize its `metrics_fun` attribute (if functions have `serialize` method), instead of saving `ss` apart from `regressor`. How do you think?

Reply:
That would be nice!
Can you implement to extend `Regressor`/`Classifier` class to serialize its `metrics_fun` attributes, and serializable `StandardScaler`?
I think it is nice if `Scaler` class (something used in preprocessing/postprocessing) can be serialized.
I think this will be another PR.

Reply:
I am not sure `Scaler` class should be an instance of `Link` or not.

Reply:
If I understand correctly, `Serializer` and `Deserializer` can serialize any object that implements `serialize` method.

Reply:
For example, `SerialIterator` has `serialize` method to perpetuate its status (cf. [here](https://github.com/chainer/chainer/blob/master/chainer/iterators/serial_iterator.py#L122)).

Reply:
todo future